### PR TITLE
FEAT: Add expression as a dict [Fields calculator]

### PR DESCRIPTION
**Pull request by gmalinve**

### CODE REVIEW

File: pyaedt/generic/constants.py [hunk 1]
```diff
@@ -1089,3 +1090,16 @@ class SymbolStyle(object):
         "HorizontalLeftTriangle",
         "HorizontalRightTriangle",
     )
+
+
+class ExpressionsCatalog(Enum):
```
Comment (Samuelopez-ansys) [inline pyaedt/generic/constants.py:1095]: Maybe ExpressionTemplate could be a better name?

File: pyaedt/modules/fields_calculator.py [hunk 2]
```diff
@@ -342,13 +360,51 @@ def load_expression_file(self, input_file):
...
+                new_expression = self.validate_expression(new_expression_props)
+                if new_expression:
+                    self.expression_catalog.update(new_expression)
 
         return self.expression_catalog
 
+    @pyaedt_function_handler()
+    def validate_expression(self, expression):
+        """Validate expression file against the schema. The default schema
+        can be found in ``pyaedt/misc/fields_calculator.schema.json``.
```
Comment (SMoraisAnsys) [inline pyaedt/modules/fields_calculator.py:376]:
```suggestion
        """Validate expression file against the schema.
        
        The default schema can be found in ``pyaedt/misc/fields_calculator.schema.json``.
```

File: pyaedt/modules/fields_calculator.py [hunk 3]
```diff
@@ -342,13 +360,51 @@ def load_expression_file(self, input_file):
...
+            Expression defined as a dictionary.
+
+        Returns
+        -------
+        dict or bool
+            Expression if the input expression is valid, ``False`` otherwise.
+        """
+
+        if not isinstance(expression, dict):
+            self.__app.logger.warning("Incorrect data type.")
```
Comment (SMoraisAnsys) [inline pyaedt/modules/fields_calculator.py:390]:
```suggestion
            self.__app.logger.error("Incorrect data type.")
```

File: pyaedt/modules/fields_calculator.py [hunk 4]
```diff
@@ -71,10 +84,15 @@ def add_expression(self, calculation, assignment, name=None):
...
 
         if calculation not in self.expression_names:
-            self.__app.logger.error("Calculation is not available.")
-            return False
-
-        expression_info = copy.deepcopy(self.expression_catalog[calculation])
+            if isinstance(calculation, dict):
+                expression_info = self.validate_expression(calculation)
+                if not expression_info:
+                    return False
```
[the remark this answers is not in the document]
Comment (SMoraisAnsys) [reply pyaedt/modules/fields_calculator.py:90]: This is resolved but the return False call stiff has no error logged to it.

**Review by SMoraisAnsys — DISMISSED** [review]
LGTM, I left minor comments

**Pull request merged**